hspec-core/src/Test/Hspec/Core/Formatters/Internal.hs

```diff
@@ -225,9 +226,27 @@ clearTransientOutput = do
 
 -- | Append some output to the report.
 write :: String -> FormatM ()
-write s = do
+write str = do
+  useColor <- getConfig formatConfigUseColor
```

Comment:
an alternative would be to check `useColor` in `withColor_`, not sure which is nicer

Reply:
I think that is what I would do.  I may refactor it.